coreclient/src/utils/persistence.rs

```
@@ -114,6 +120,74 @@ fn client_db_name(user_id: &UserId) -> String {
     format!("{}@{}.db", user_id.uuid(), user_id.domain())
 }
 
+pub async fn export_client_database(db_path: &str, user_id: &UserId) -> Result<Vec<u8>> {
+    let client_db_name = client_db_name(user_id);
+
+    // Commit the WAL to the database file
+    let db_url = format!("sqlite://{db_path}/{client_db_name}");
+    let opts: SqliteConnectOptions = db_url.parse()?;
+    let opts = opts
+        .journal_mode(SqliteJournalMode::Wal)
+        .create_if_missing(true);
+    let pool = SqlitePoolOptions::default().connect_with(opts).await?;
+    sqlx::query("VACUUM").execute(&pool).await?;
+    pool.close().await;
+
+    // Create a tar archive of the database
+    let mut data = Vec::new();
+    let enc = GzEncoder::new(&mut data, Compression::default());
+    let mut tar = tar::Builder::new(enc);
+
+    let client_db_path = format!("{db_path}/{client_db_name}");
+    let content = fs::read(client_db_path)?;
+    let mut header = tar::Header::new_gnu();
+    header.set_size(content.len().try_into().expect("usize overflow"));
```

Could you throw an error here instead?